src/server/vnparse.c

```diff
@@ -2486,6 +2481,139 @@ typedef struct resc_limit_entry {
 	resc_limit_t	*resc;
 } rl_entry;
 
+/* Reusing ATR_VFLAG_HOOK to mean that the resource key value pair was found
+ * in in execvnode
+ */
+#define IN_EXECVNODE_FLAG	ATR_VFLAG_HOOK
+
+#if !(defined(PBS_MOM) || defined(PBS_PYTHON))
+/**
+ * @brief
+ * compare lexicographically the names of resources in the resource list contained in
+ * two resc_limit_t structures
+ *
+ * @param[in]	left - the left resource limit.
+ * @param[in]	right - the right resource limit.
+ *
+ * @return int
+ * @retval -1	- left < right
+ * @retval 0	- left = right
+ * @retval 1	- left > right
+ */
+static int
+resc_limit_list_cmp_name(resc_limit_t *left, resc_limit_t *right)
+{
+	resource *pres_l, *pres_r;
+
+	if (left->rl_ncpus && !right->rl_ncpus)
+		return 1;
+	if (!left->rl_ncpus && right->rl_ncpus)
+		return -1;
+
+	if (left->rl_ssi && !right->rl_ssi)
+		return 1;
+	if (!left->rl_ssi && right->rl_ssi)
+		return -1;
+
+	if (left->rl_mem && !right->rl_mem)
+		return 1;
+	if (!left->rl_mem && right->rl_mem)
+		return -1;
+
+	if (left->rl_vmem && !right->rl_vmem)
+		return 1;
+	if (!left->rl_vmem && right->rl_vmem)
+		return -1;
+
+	if (left->rl_naccels && !right->rl_naccels)
+		return 1;
+	if (!left->rl_naccels && right->rl_naccels)
+		return -1;
+
+	if (left->rl_accel_mem && !right->rl_accel_mem)
+		return 1;
+	if (!left->rl_accel_mem && right->rl_accel_mem)
+		return -1;
+
+	for (pres_l = (resource *)GET_NEXT(left->rl_oth_res),
+		pres_r = (resource *)GET_NEXT(right->rl_oth_res);
+		pres_l && pres_r;
+		pres_l = (resource *)GET_NEXT(pres_l->rs_link),
+		pres_r = (resource *)GET_NEXT(pres_r->rs_link)) {
+		int cmp_res;
+		if ((cmp_res = strcasecmp(pres_l->rs_defin->rs_name, pres_r->rs_defin->rs_name)))
+			return cmp_res;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief
+ * compare the values of resources in the resource list contained in
+ * two resc_limit_t structures
+ *
+ * @param[in]	left - the left resource limit.
+ * @param[in]	right - the right resource limit.
+ *
+ * @return int
+ * @retval -1	- left < right
+ * @retval 0	- left = right
+ * @retval 1	- left > right
+ */
+static int
+resc_limit_list_cmp_val(resc_limit_t *left, resc_limit_t *right)
```

Comment:
Check for NULL-ness of 'left' and 'right'.

Reply:
Yes, please still do the check.

Reply:
Done, see commit https://github.com/PBSPro/pbspro/pull/1452/commits/6151018baa0b74a7f013bf2e272e4ce090374365